src/Voron/Data/DirectAddScope.cs

```diff
@@ -5,35 +5,45 @@ namespace Voron.Data
 {
     public unsafe class DirectAddScope : IDisposable
     {
-        private readonly ITree _tree;
+        private readonly string _treeName;
         private uint _usage;
+#if DEBUG
         private string _allocationStacktrace = null;
-
+#endif
         public byte* Ptr;
 
-        public DirectAddScope(ITree tree)
+        public DirectAddScope(string treeName)
         {
             _usage = 0;
-            _tree = tree;
+            _treeName = treeName;
             Ptr = null;
         }
 
         public DirectAddScope Open(byte* writePos)
         {
-            if (_usage++ > 0)
-                ThrowScopeAlreadyOpen(_tree, _allocationStacktrace);
-
-            Ptr = writePos;
+            if (_usage++ <= 0)
+            {
+                Ptr = writePos;
 #if DEBUG
-            // uncomment for debugging purposes only
-            //_allocationStacktrace = Environment.StackTrace;
+                // uncomment for debugging purposes only
+                //_allocationStacktrace = Environment.StackTrace;
 #endif
+            }
+            else
+            {
+#if DEBUG
+                ThrowScopeAlreadyOpen(_treeName, _allocationStacktrace);
+#else
+                ThrowScopeAlreadyOpen(_treeName, null);
+#endif
+            }
+
             return this;
         }
 
-        private static void ThrowScopeAlreadyOpen(ITree tree, string previousOpenStacktrace)
+        private static void ThrowScopeAlreadyOpen(string treeName, string previousOpenStacktrace)
```

Comment:
Sorry, I misunderstood. I think we can get rid of it completely. If we got this error then we'll have to debug it anyway. It's very little sense of keeping it.